suitcase/tiff/__init__.py

```diff
@@ -1,25 +1,312 @@
-from ._version import get_versions
-__version__ = get_versions()['version']
-del get_versions
-
-
-# Suitcase subpackages must follow strict naming and interface conventions. The
+# suitcase subpackages must follow strict naming and interface conventions. The
 # public API should include some subset of the following. Any functions not
 # implemented should be omitted, rather than included and made to raise
 # NotImplementError, so that a client importing this library can immediately
 # know which portions of the suitcase API it supports without calling any
 # functions.
 #
-# def export(...)
-#     ...
-#
-#
-# def ingest(...):
-#     ...
-#
-#
-# def reflect(...):
-#     ...
-#
-#
-# handlers = []
+from collections import defaultdict
+from pathlib import Path
+import json
+import tifffile
+import event_model
+import numpy
+import suitcase.utils
+import collections
+from ._version import get_versions
+
+__version__ = get_versions()['version']
+del get_versions
+
+
+class SuitcaseTiffError(Exception):
+    ...
+
+
+class NonSupportedDataShape(SuitcaseTiffError):
+    '''used to indicate that non-supported data type is being saved
+    '''
+    ...
+
+
+def export(gen, directory, file_prefix='', **kwargs):
+    """
+    Export a stream of documents to tiff file(s) and one JSON file of metadata.
+
+    Creates {filepath}_meta.json and then {filepath}_{stream_name}.tiff
+    for every Event stream. It can also serialize the data to any file handle.
+
+    The structure of the json is::
+
+            {'metadata': {'start': start_doc, 'stop': stop_doc,
```

Comment:
Reminder of our discussion at the end of the day last Friday: What if I have a stream named 'metadata'? I think scoping the stream data under 'streams' is reasonable.

Reply:
thanks for the reminder. this is fixed